js/raster_layer_editor.ts

```diff
@@ -0,0 +1,345 @@
+                    name="color-ramp"
+                    value="${option.value}"
+                    @click="${this.onColorRampChanged}"
+                    ?checked="${this.colorRamp === option.value}"
+                />
+                <label class="legacy-text">${option.label}</label>
+            </span>
+        `;
+    }
+
+    private onColorRampChanged(event: Event): void {
+        this.colorRamp = (event.target as HTMLInputElement).value;
+    }
+
+    private renderBandSelection(value: string): TemplateResult {
+        return renderSelect(this.bandNames, value, this.onBandSelectionChanged);
+    }
+
+    private onRefreshButtonClicked(event: Event): void {
+        event.stopImmediatePropagation();
+        this.calculateBandStats();
+    }
+
+    private onOpacityChanged(event: Event): void {
+        this.opacity = (event.target as HTMLInputElement).valueAsNumber;
+    }
+
+    private onGammaChanged(event: Event): void {
+        this.gamma = (event.target as HTMLInputElement).valueAsNumber;
+    }
+
+    private onMinTextChanged(event: Event): void {
+        this.minValue = (event.target as HTMLInputElement).valueAsNumber;
+    }
+
+    private onMaxTextChanged(event: Event): void {
+        this.maxValue = (event.target as HTMLInputElement).valueAsNumber;
+    }
+
+    private calculateBandStats(): void {
+        if (this.stretch === "custom") {
+            return;
+        }
+        this.minValue = undefined;
+        this.maxValue = undefined;
+        this.dispatchEvent(
+            new CustomEvent("calculate-band-stats", {
+                bubbles: true,
+                composed: true,
+            })
+        );
+    }
+
+    private onStretchChanged(event: Event): void {
+        this.stretch = (event.target as HTMLInputElement).value;
+        this.minAndMaxValuesLocked = this.stretch !== "custom";
+    }
+
+    private onBandSelectionChanged(_event: Event): void {
+        this.selectedBands = this.getSelectedBands();
+    }
+
+    private onColorModelChanged(event: Event): void {
+        this.colorModel = (event.target as HTMLInputElement).value;
+    }
+
+    updated(changedProperties: PropertyValues<RasterLayerEditor>): void {
+        super.updated(changedProperties);
+
+        if (changedProperties.has("colorModel")) {
+            if (this.colorModel === ColorModel.Gray) {
+                this.selectedBands = Array.from([this.bandNames[0]]);
+            } else if (this.colorModel == ColorModel.RGB) {
+                this.selectedBands = Array.from([
+                    this.bandNames[0],
+                    this.bandNames[0],
+                    this.bandNames[0],
+                ]);
+            }
+        }
+
+        if (changedProperties.has("selectedBands") ||
+            changedProperties.has("stretch")) {
+            this.calculateBandStats();
+        }
+    }
+
+    private getSelectedBands(): Array<string> {
+        const container = this.renderRoot?.querySelector(
```

Comment:
I believe we can simplify this by adding a new property, `bandSelects`, decorated with`@queryAll("#band-selection select")`:  https://lit.dev/docs/api/decorators/#queryAll